README.md

```diff
@@ -37,29 +37,34 @@ MSRV 1.51.0 due to use of const generics
 
 ## Usage
 
+The `multihash` crate exposes a basic data structure for encoding and decoding multihash.
+It does not provide any hashing functionality itself.
+`Multihash` uses const-generics to define the internal buffer size.
+You should set this to the maximum size of the digest you want to support.
+
 ```rust
-use multihash::{Code, MultihashDigest};
+use multihash::Multihash;
 
 fn main() {
-    let hash = Code::Sha2_256.digest(b"my hash");
+    let hash = Multihash::<64>::wrap(0x12, b"my digest");
```

Comment:
Nitpick (won't prevent a merge). I prefer that we don't promote using magic numbers:

```rust
use multihash::Multihash;

const SHA2_256: u64 = 0x12;

fn main() {
    let hash = Multihash::<64>::wrap(SHA2_256, b"my digest");
    println!("{:?}", hash);
}
```